compiler/rustc_session/src/parse.rs

```diff
@@ -184,6 +185,8 @@ pub fn add_feature_diagnostics_for_issue(
         } else {
             err.subdiagnostic(FeatureDiagnosticHelp { feature });
         }
+
+        err.subdiagnostic(SuggestUpgradeCompiler::new());
```

Comment:
This should be conditional -- could you try to parse the date and compare it to now or something?

Reply:
Definitely can-do, the date format seems to be a triple: `YYYY-MM-DD`, so I can definitely parse that. As for getting the current date, am I allowed to import the `chrono` crate, or something? Not sure how to handle this, the standard library doesn't seem to have great datetime facilities

Reply:
I'd be very.. careful about detecting dates like this. Would definitely have to be turned off for the test suite but even outside, it gives off bad vibes. If prefer just always showing it, even if it is the latest version.